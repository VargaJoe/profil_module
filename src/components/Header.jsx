import React, { Component } from 'react';

class Header extends Component {
	render () {
		return (
			<header className="header">
				<div className="wrapp">
					<div className="header__search">
						<input type="text" />
						<button>
							<i className="fi flaticon-magnifier"></i>
						</button>
					</div>
				</div>
			</header>
		);
	}
};

export default Header;
