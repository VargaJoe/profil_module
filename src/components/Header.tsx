import * as React						from 'react';
import { connect }                      from 'react-redux';
import { IODataParams } 				from '@sensenet/client-core';
import { User } 						from '@sensenet/default-content-types';
import { search } 						from '../reducers/users';

interface Props {
    search: Function;
}

class Header extends React.Component<any, any> {

	constructor(props: Props) {
        super(props);
        this.state = {},

        this.handleChange =        this.handleChange.bind(this);
	}
	
    public handleChange = (e:  any) => {
		if ( e.target.value.length > 2 ) {
			console.log(e.target.value);
		}
	}
	
	render () {
		return (
			<header className="sn_header">
				<div className="sn_wrapp">
					<div className="sn_header__search">
						<input type="text"  onChange={this.handleChange}/>
						<button>
							<i className="fi flaticon-magnifier" />
						</button>
					</div>
				</div>
			</header>
		);
	}
}

export const mapStateToProps = (state: any, match: any) => {
    return {};
};

export default connect(
    mapStateToProps,
    (dispatch) => ({
        search:       (path: string, options: IODataParams<User>) => dispatch(search( path, options )),
    })
)(Header as any);