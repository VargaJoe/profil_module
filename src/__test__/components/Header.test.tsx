import * as React 						from 'react';
import Header 							from '../../components/Header';
import { 
	configure, 
    shallow,
    mount } 							from 'enzyme';
import * as Adapter 					from 'enzyme-adapter-react-16';
import toJson 				      		from 'enzyme-to-json';

configure( {adapter: new Adapter()} );

describe('<Header /> shallow rendering', () => {
    
    const header = shallow(<Header />); 
    
    // test Snapshot 
	it('Match to snapshot', () => {
		expect(toJson(header)).toMatchSnapshot();
    });
    
});
