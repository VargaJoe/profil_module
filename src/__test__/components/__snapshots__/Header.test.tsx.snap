// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> shallow rendering Match to snapshot 1`] = `
<header
  className="sn_header"
>
  <div
    className="sn_wrapp"
  >
    <div
      className="sn_header__search"
    >
      <input
        type="text"
      />
      <button>
        <i
          className="fi flaticon-magnifier"
        />
      </button>
    </div>
  </div>
</header>
`;
